A1-http-connect-proxy-support.md

```diff
@@ -1,222 +1,265 @@
-
-- We will support a new type of hook called a *proxy mapper*, which
-  offers two points at which proxies can be selected:
-  - Right before the server name is resolved, the proxy mapper can be used
-    to programmatically override the name that will be resolved.  It
-    will also be able to set the internal state to trigger use of the
-    HTTP CONNECT handshaker.
-  - Right before we connect to the target address, the proxy mapper can
-    be used to programmatically override the address that we will
-    connect to.  It will also be able to set the internal state to trigger
-    use of the HTTP CONNECT handshaker.
+-   We will have some internal state (e.g., a channel arg in C-core) that both
+    triggers the use of the HTTP CONNECT handshaker and specifies the argument
+    to be used in the CONNECT request.
+
+-   We will support a new type of hook called a *proxy mapper*, which offers two
+    points at which proxies can be selected:
+
+    -   Right before the server name is resolved, the proxy mapper can be used
+        to programmatically override the name that will be resolved. It will
+        also be able to set the internal state to trigger use of the HTTP
+        CONNECT handshaker.
+    -   Right before we connect to the target address, the proxy mapper can be
+        used to programmatically override the address that we will connect to.
+        It will also be able to set the internal state to trigger use of the
+        HTTP CONNECT handshaker.
 
 ### Addressing Use Cases
 
-This new functionality will be used to address the three use-cases
-described above, as follows.
+This new functionality will be used to address the three use-cases described
+above, as follows.
 
 #### Case 1
 
-Case 1 will be addressed using the proxy mapper hook for overriding the
-server name to resolve.  Sites that may want to send only external servers
-through the proxy can implement their own proxy mapper to do this.
-We will provide a default proxy mapper implementation that looks for the
-existence of the `http_proxy` environment variable (or equavalent
-signal).  If that signal is present, it will do the following:
-- Ask the resolver to resolve the proxy name instead of the server name
-  requested via the client API.  (Note that this will cause us to
-  create a single subchannel to the proxy, as opposed to one
-  subchannel for each backend server.)
-- Add the state described above triggering the use of the HTTP CONNECT
-  handshaker, with the argument set to the server name requested via
-  the client API.
+Case 1 will be addressed using the proxy mapper hook for overriding the server
+name to resolve. Sites that may want to send only external servers through the
+proxy can implement their own proxy mapper to do this. We will provide a default
+proxy mapper implementation that looks for the existence of the `http_proxy`
+environment variable (or equavalent signal). If that signal is present, it will
```

Comment:
s/equavalent/equivalent/

Reply:
Not proceeding with changes to this doc anymore